test/SpagoSpec.hs

```diff
@@ -0,0 +1,158 @@
+module SpagoSpec (spec) where
+
+import           Control.Concurrent  (threadDelay)
+import           Control.Monad.Extra (whenM)
+import           Prelude             hiding (FilePath)
+import           Test.Hspec          (Spec, around_, beforeAll, describe, it,
+                                      shouldBe)
+import           Turtle              (FilePath, cp, mkdir, mv, readTextFile, rm,
+                                      rmtree, testdir, writeTextFile)
+import           Utils               (checkFixture, runFor, shouldBeFailure,
+                                      shouldBeFailureOutput, shouldBeSuccess,
+                                      shouldBeSuccessOutput, spago, withCwd)
+
+testDir :: FilePath
+testDir = "test/spago-test"
+
+clean :: IO ()
+clean = do
+  whenM (testdir testDir) (rmtree testDir)
+  mkdir testDir
+
+spec :: Spec
+spec = beforeAll clean $ around_ (withCwd testDir) $ do
```

Comment:
In the psc-package spec we clean both before and after but here only before, is it accidental or there's a reason for it?

Reply:
It was intentional, but I don't mind changing it:
- `test/psc-package-local-test` is checked into git, so it gets cleaned afterwards to avoid unwanted changes showing up in the git diff.
- `test/spago-test` is not checked into git (the folder is in the `.gitignore`), so we can leave the crumbs from the tests there for possible debugging purposes.

Reply:
Oh right, let's keep it like this 👍